src/components/visualizations/table/utils/dataTransformation.ts

```diff
@@ -158,27 +139,33 @@ export function getTotalsWithData(
     }, []);
 }
 
-export function validateTableProportions(headers: TableHeader[], rows: TableRow[]): void {
+export function validateTableProportions(headers: IMappingHeader[], rows: TableRow[]): void {
     invariant(
         rows.length === 0 || headers.length === rows[0].length,
         'Number of table columns must be equal to number of table headers'
     );
 }
 
-export function getIntersectionForDrilling(afm: AFM.IAfm, header: TableHeader): IDrillIntersection {
-    return isAttributeTableHeader(header)
-        ? {
-            id: header.identifier,
-            identifier: header.identifier,
-            uri: header.uri,
-            title: header.name
-        }
-        : {
-            id: header.localIdentifier,
+export function getIntersectionForDrilling(afm: AFM.IAfm, header: IMappingHeader): IDrillIntersection {
+    if (isMappingHeaderAttribute(header)) {
+        return {
+            id: header.attributeHeader.identifier,
+            identifier: header.attributeHeader.identifier,
+            uri: header.attributeHeader.uri,
+            title: getMappingHeaderName(header)
+        };
+    }
+
+    if (isMappingHeaderMeasureItem(header)) {
+        return {
+            id: header.measureHeaderItem.localIdentifier,
             identifier: '',
-            uri: get(getMeasureUriOrIdentifier(afm, header.localIdentifier), 'uri') as string,
-            title: header.name
+            uri: get<string>(getMasterOrDerivedUriAndIdentifier(afm, header.measureHeaderItem.localIdentifier), 'uri'),
+            title: getMappingHeaderName(header)
         };
+    }
+
+    throw new Error(`Unknown mapping header type ${Object.keys(header)}`);
```

Comment:
what about `TotalHeaderItems` ? which are in same collection as attribute & measure headers

Reply:
We do not allow drill on totals